lib/pbench/server/api/resources/query_apis/__init__.py

```diff
@@ -754,6 +755,10 @@ def _post(
                     raise_on_error=False,
                 )
                 report = self._analyze_bulk(results)
+            except UnauthorizedAdminAccess as e:
+                raise APIAbort(e.http_status, str(e))
+            except MissingParameters as e:
+                raise APIAbort(e.http_status, str(e))
```

Comment:
Do these exceptions actually get raised (and caught here)?  In the call to `streaming_bulk()` at line 751 we specify `raise_on_exception` and `raise_on_error` to be `False`...those don't prevent exceptions (e.g., from `generate_actions()`) from reaching here?

Reply:
They're raised by `generate_actions`, which `dataset_update.py` uses to perform the extra validation of query parameters and `ADMIN` access. The fact that `streaming_bulk` doesn't report Elasticsearch errors as exceptions doesn't affect this; but of course it has to be inside the `try` because `generate_actions` is a generator that won't be activated (to validate and raise these exceptions) until the first iteration from `streaming_bulk`.

Reply:
I just wanted to understand whether `streaming_bulk()` would catch the exceptions raised by `generate_actions()`, or whether they would make it to these new `except` clauses.

I just looked at the Cobertura report (which is what I should have done in the first place), and, indeed, we have test coverage which actually executes those lines, so bravo, and I will wander away, now.